test/private2.js

```diff
@@ -0,0 +1,30 @@
+const createSSB = require('./util/create-ssb')
+const tape = require('tape')
+const { promisify } = require('util')
+
+tape('partial index', async (t) => {
+  const name = `test-ssb-partial-index-${Date.now()}`
+
+  t.plan(6)
+
+  const ssb = createSSB(name, { temp: false })
+  const { key } = await promisify(ssb.publish)({ type: 'test' })
+  t.pass('published')
+
+  await promisify(ssb.rebuild)()
+  t.pass('rebuilt')
+
+  await promisify(ssb.close)()
+  t.pass('closed')
+
+  const ssb2 = createSSB(name, { temp: false }, [require('ssb-private2')])
+
+  const result = await promisify(ssb2.get)(key)
+  t.ok(result)
+
+  await promisify(ssb2.publish)({ type: 'test' })
+  t.pass('published again')
+
+  await promisify(ssb2.close)()
+  t.pass('closed again')
```

Comment:
I don't understand what this is testing exactly

Reply:
I don't completely understand why, but this was able to give us a minimal reproduction of the problem that we were bumping into with `whakapapa-ora/desktop/rebuild.temp.js`.

Reply:
oh with the locked up state?

Reply:
I've left a note in that file so people looking at it have some context